### DISCUSSION TITLE
Dropping a transaction based on its name and duration

### DISCUSSION THREAD

**Discussion by LorenzoScebba** [getsentry/sentry-java#2213] [Q&A] [answered]
Hi,

On our project we have a pretty aggressive scheduler running every 5 seconds, the scheduler starts a transaction called "Process".

Since the transaction could last from a couple of milliseconds up to tens of seconds, we'd like to send only the "problematic" transaction to sentry (the ones where the transaction duration is >= 1sec).

Is this achievable in any way?

Thanks,
Lorenzo

**Comment by adinauer** [comment, marked as answer]
Hello @LorenzoScebba, you could write your own `EventProcessor` and implement the `SentryTransaction process(@NotNull SentryTransaction transaction, @NotNull Hint hint)` method. In it you could calculate the difference between `transaction.getTimestamp()` and `transaction.getStartTimestamp()`, then return `null` to drop the transaction.

Here's what this might look like:
```
public class MyEventProcessor implements EventProcessor {

  @Override
  public @Nullable SentryTransaction process(@NotNull SentryTransaction transaction, @NotNull Hint hint) {
    @Nullable Double end = transaction.getTimestamp();

    if (end != null && "Process".equals(transaction.getTransaction())) {
      @NotNull Double start = transaction.getStartTimestamp();
      @NotNull Double difference = end - start;
      if (difference < 10.0) { // threshold in seconds
        return null;
      }
    }

    return transaction;
  }
}
```

You have to add the `EventProcessor` on `init` using `options.addEventProcessor(new MyEventProcessor())`.

**Reply by LorenzoScebba** [reply]
Hi @adinauer, 

Thanks a lot for the snippet! We'll definitly try this out.

Thanks again!
